### PR TITLE
Luna Streaming 3.1 upgrade guide

### PR DESCRIPTION
**Pull request by mendonk**
* Impacts and considerations for upgrading from LS 2.10 to 3.1
* Upgrade guide for LS 2.10 -> 3.1

Preview:
[View draft](https://d5rxiv0do0q3v.cloudfront.net/upgrade-guide/luna-streaming/3.1-3.x/install-upgrade/upgrade-guide.html)

### CODE REVIEW

**Comment by plpesvc-ds** [comment]
Antora site build successful! :white_check_mark:
Deploying draft.
Deployment successful! [View draft](https://d5rxiv0do0q3v.cloudfront.net/upgrade-guide/luna-streaming/3.1-3.x/index.html)

**Comment by plpesvc-ds** [comment]
Antora site build successful! :white_check_mark:
Deploying draft.
Deployment successful! [View draft](https://d5rxiv0do0q3v.cloudfront.net/upgrade-guide/luna-streaming/3.1-3.x/index.html)

**Comment by plpesvc-ds** [comment]
Antora site build failure! :x:
>

**Comment by plpesvc-ds** [comment]
Antora site build successful! :white_check_mark:
Deploying draft.
Deployment successful! [View draft](https://d5rxiv0do0q3v.cloudfront.net/upgrade-guide/luna-streaming/3.1-3.x/index.html)

**Comment by plpesvc-ds** [comment]
Antora site build successful! :white_check_mark:
Deploying draft.
Deployment successful! [View draft](https://d5rxiv0do0q3v.cloudfront.net/upgrade-guide/luna-streaming/3.1-3.x/index.html)

**Comment by plpesvc-ds** [comment]
Antora site build successful! :white_check_mark:
Deploying draft.
Deployment successful! [View draft](https://d5rxiv0do0q3v.cloudfront.net/upgrade-guide/luna-streaming/3.1-3.x/index.html)

File: modules/install-upgrade/pages/upgrade-guide.adoc [hunk 1]
```diff
@@ -0,0 +1,554 @@
...
+        loadBalancerSheddingConditionHitCountThreshold: 1e
+----
++
+. To upgrade your existing Pulsar installation, run the following Helm command.
+This command assumes the default `pulsar` namespace. If you are using a different namespace, replace `pulsar` with your namespace.
+The `--wait` flag ensures that Helm waits until all pods are ready before completing the upgrade.
++
+[source,bash]
+----
+helm upgrade pulsar --namespace pulsar --values current-values.yaml --wait
```
Comment (nikhil-ctds) [inline modules/install-upgrade/pages/upgrade-guide.adoc]: Please update the `helm upgrade` command. Please refer [here](https://docs.google.com/document/d/1mSbx0ig-QPTRe23gRqlqNKvptX5qwrugCAH-TGOYlsY/edit#heading=h.3zbqqcbvk8ok).

File: modules/install-upgrade/pages/upgrade-guide.adoc [hunk 2]
```diff
@@ -0,0 +1,554 @@
...
+    brokerDeduplicationEnabled: "false"
+----
++
+. To upgrade your existing Pulsar installation, run the following Helm command.
+This command assumes the default `pulsar` namespace. If you are using a different namespace, replace `pulsar` with your namespace.
+The `--wait` flag ensures that Helm waits until all pods are ready before completing the upgrade.
++
+[source,bash]
+----
+helm upgrade pulsar datastax-pulsar/pulsar --namespace pulsar --values current-values.yaml --wait
```
Comment (nikhil-ctds) [inline modules/install-upgrade/pages/upgrade-guide.adoc]: Please update the `helm upgrade` command. Please refer [here](https://docs.google.com/document/d/1mSbx0ig-QPTRe23gRqlqNKvptX5qwrugCAH-TGOYlsY/edit#heading=h.5n3gpe4duaii).

File: modules/install-upgrade/pages/upgrade-guide.adoc [hunk 3]
```diff
@@ -0,0 +1,554 @@
...
+Deploying Luna Streaming on Kubernetes with KAAP (Kubernetes Autoscaling for Apache Pulsar) Operator is a common method for running Pulsar in a cloud-native environment.
+
+For more information, see the xref:kaap-operator::index.adoc[KAAP documentation].
+
+. Back up your existing Pulsar data and configurations to prevent data loss.
+. To save your current Helm release configuration, run the following command:
++
+[source,bash]
+----
+helm get values > pulsar-backup-values.yaml
```
Comment (nikhil-ctds) [inline modules/install-upgrade/pages/upgrade-guide.adoc]: Helm get values requires release name. Please update the command with below given command:
`helm get values <release-name> > pulsar-backup-values.yaml`

File: modules/install-upgrade/pages/upgrade-guide.adoc [hunk 4]
```diff
@@ -0,0 +1,554 @@
...
+The Helm chart for Luna Streaming is available in the https://github.com/datastax/pulsar-helm-chart/blob/master/helm-chart-sources/pulsar/values.yaml[Helm chart sources] repository.
+
+Deploying Luna Streaming on Kubernetes using the DataStax Helm chart is another common method for running Pulsar in a cloud-native environment.
+
+. Back up your existing Pulsar data and configurations to prevent data loss.
+. To save your current Helm release configuration, run the following command:
++
+[source,bash]
+----
+helm get values > pulsar-backup-values.yaml
```
Comment (nikhil-ctds) [inline modules/install-upgrade/pages/upgrade-guide.adoc]: Helm get values requires release name. Please update the command with below given command:
`helm get values <release-name> > pulsar-backup-values.yaml`

File: modules/install-upgrade/pages/upgrade-guide.adoc [hunk 5]
```diff
@@ -0,0 +1,554 @@
...
+[source,bash]
+----
+kubectl get pods --namespace pulsar
+----
+=
+. Check the logs for any issues:
++
+[source,bash]
+----
+kubectl logs -n
```
Comment (nikhil-ctds) [inline modules/install-upgrade/pages/upgrade-guide.adoc]: `kubectl logs` command is not correct. Please update with below given command:
`kubectl logs <pod-name> -n <namespace>`

File: modules/install-upgrade/pages/upgrade-guide.adoc [hunk 6]
```diff
@@ -0,0 +1,554 @@
...
+[source,bash]
+----
+kubectl get pods --namespace pulsar
+----
+=
+. To check the logs for any issues, run the following command:
++
+[source,bash]
+----
+kubectl logs -n
```
Comment (nikhil-ctds) [inline modules/install-upgrade/pages/upgrade-guide.adoc]: `kubectl logs` command is not correct. Please update with below given command:
`kubectl logs <pod-name> -n <namespace>`

File: modules/install-upgrade/pages/upgrade-guide.adoc [hunk 7]
```diff
@@ -0,0 +1,554 @@
...
+helm upgrade pulsar --namespace pulsar --values current-values.yaml --wait
+----
++
+. Check the status of the pods to ensure they are running correctly:
++
+[source,bash]
+----
+kubectl get pods --namespace pulsar
+----
+=
```
Comment (nikhil-ctds) [inline modules/install-upgrade/pages/upgrade-guide.adoc]: it seems to be a typo and `=` shouldn't be present.

File: modules/install-upgrade/pages/upgrade-guide.adoc [hunk 8]
```diff
@@ -0,0 +1,554 @@
...
+helm upgrade pulsar datastax-pulsar/pulsar --namespace pulsar --values current-values.yaml --wait
+----
++
+. To check the status of the pods to ensure they are running correctly, run the following command:
++
+[source,bash]
+----
+kubectl get pods --namespace pulsar
+----
+=
```
Comment (nikhil-ctds) [inline modules/install-upgrade/pages/upgrade-guide.adoc]: it seems to be a typo and `=` shouldn't be present.

File: modules/install-upgrade/pages/upgrade-guide.adoc [hunk 9]
```diff
@@ -0,0 +1,554 @@
...
+	at org.apache.bookkeeper.bookie.storage.ldb.KeyValueStorageRocksDB.<init>(KeyValueStorageRocksDB.java:197) ~[com.datastax.oss-bookkeeper-server-4.14.5.1.0.2.jar:4.14.5.1.0.2]
+	... 13 more
+----
+=====
+
+== Upgrade procedure
+
+Luna Streaming can be deployed on Bare metal, Docker, and Kubernetes.
+
+=== Bare metal deployment
```
Comment (mukesh-ctds) [inline modules/install-upgrade/pages/upgrade-guide.adoc]: I think we just need Kubernetes mode of deployment the doc as per @pgier. We should remove this section for bare metal.

File: modules/install-upgrade/pages/upgrade-guide.adoc [hunk 10]
```diff
@@ -0,0 +1,554 @@
...
+
+Bare metal deployment installs Luna Streaming directly on physical servers.
+
+For detailed instructions on upgrading Pulsar, see the https://pulsar.apache.org/docs/next/administration-upgrade/[Pulsar Upgrade Guide].
+
+// known issues?
+
+=== Docker deployment
```
Comment (mukesh-ctds) [inline modules/install-upgrade/pages/upgrade-guide.adoc]: I think we just need Kubernetes mode of deployment the doc as per @pgier. We should remove this section for Docker deployment.

**Comment by plpesvc-ds** [comment]
Antora site build successful! :white_check_mark:
Deploying draft.
Deployment successful! [View draft](https://d5rxiv0do0q3v.cloudfront.net/upgrade-guide/luna-streaming/3.1-3.x/index.html)

**Comment by plpesvc-ds** [comment]
The draft build for this pull request has been cleaned up due to being stale.

Draft builds 2 weeks old or older were automatically removed.

You can rebuild the draft at any time by pushing a new commit to this pull request:

```
git checkout upgrade-guide
git commit --allow-empty -m 'rebuild draft'
git push origin upgrade-guide
```

**Comment by plpesvc-ds** [comment]
Build successful! :white_check_mark:
Deploying draft.
Deploy successful! [View draft](https://d5rxiv0do0q3v.cloudfront.net/upgrade-guide/luna-streaming/3.1-3.x/index.html)

**Comment by eric-schneider** [comment]
SME approval given in [Slack](https://datastax.slack.com/archives/C01DJ4M36SV/p1728977494719429?thread_ts=1727722347.299409&cid=C01DJ4M36SV). We will forgo a formal style review for the time being, but I've created [LS-1305](https://datastax.jira.com/browse/LS-1305) revisit this page in the future.

[LS-1305]: https://datastax.jira.com/browse/LS-1305?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by plpesvc-ds** [comment]
Build successful! :white_check_mark:
Deploying draft.
Deploy successful! [View draft](https://d5rxiv0do0q3v.cloudfront.net/upgrade-guide/luna-streaming/3.1-3.x/index.html)

**Review by eric-schneider — APPROVED** [review]

**Pull request merged**